proto/reaction.proto

```diff
@@ -124,6 +126,24 @@ message ReactionInput {
   Time addition_duration = 5;
   // For continuous synthesis, we instead specify a flow rate.
   FlowRate flow_rate = 6;
+  message AdditionDevice {
+    enum AdditionDeviceType {
+      UNSPECIFIED = 0;
+      CUSTOM = 1;
+      NONE = 2;
+      SYRINGE = 3;
+      CANNULA = 4;
+      ADDITION_FUNNEL = 5;
+    }
+    AdditionDeviceType type = 1;
+    // Device specifications, e.g., "gas-tight" for SYRINGE.
+    string details = 2;
```

Comment:
I'd suggest we remove `details` from inside this `AdditionDevice` message and add a new field below, `string addition_details = 9;` which will be a catch-all for details. Examples of what could be specified include "gas-tight" for a SYRINGE `addition_device`, information about the portion sizes for a PORTIONWISE `addition_speed`, and _how_ the reaction input was heated/cooled if `addition_temperature` is specified.

Reply:
Agreed. That sounds great. I like the comment as a whole as well.

Reply:
Are you making the edits or should I?